File: ArraysAndStrings/1.1/Golang/is_unique.go
package main

import (
	"fmt"
	"os"
	"bufio"
)

func is_unique(word string) bool {
	unique := make(map[string]bool)
	for _, char := range word {
		if _, ok := unique[string(char)]; !(ok) {
			unique[string(char)] = true
		} else {
			return false
		}
	}
	return true

}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Introduce a string")
	word, _ := reader.ReadString('\n')

	var uniqueness = is_unique(word)
	if uniqueness {
		fmt.Println("It has only unique characters")
	} else {
		fmt.Println("It does not have only unique characters")
	}
}

File: ArraysAndStrings/1.1/C/is_unique.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define HASH_TABLE_SIZE 256

int is_unique(char* word){
    int len = strlen(word);
    char* hash_table;
    hash_table = malloc(sizeof(char)*HASH_TABLE_SIZE);
    for(int i=0; i<len; i++){
        char c = word[i];
        if (hash_table[c] == c){
            free(hash_table);
            return 0;
        } else {
            hash_table[c] = c;
        }
    }
    free(hash_table);
    return 1;
}

int main(int argc, char **argv){
    if (argc < 2){
        printf("Usage: ./is_unique <word>\n");
        return -1;
    }

    char* word = argv[1];
    int valid = is_unique(word);

    if(valid){
        printf("%s has unique characters only\n", word);
    } else {
        printf("%s does not have unique characters only\n", word);
    }
    return 0;
}